init.lua

```diff
@@ -50,18 +49,18 @@ local function bind(...)
 end
 local function ScanXtar()
 	if mq.TLO.Me.XTarget() > 0 then
-		for i = 1, 20 do
+		for i = 1, mq.TLO.Me.XTargetSlots() do
 			local xName = mq.TLO.Me.XTarget(i).Name() or 'NULL'
 			local xHp = mq.TLO.Me.XTarget(i).PctHPs() or -1
 			local xId = mq.TLO.Me.XTarget(i).ID() or -1
 			local xType = mq.TLO.Me.XTarget(i).Type() or '?'
```

Comment:
You can do something like this...

```lua
local xTarg = mq.TLO.Me.XTarget(i)
local xId = xTarg.ID()
if xId > 0 then
    local xName = xTarg.Name()
    local xHp = xTarg.PctHPs()
    local xType = xTarg.Type()

    --- continue on..
```

Note that with xtarg as a local variable you probably don't need all these other ones and can just use the accessors directly.